### PR TITLE
Remove dead code and manage dependencies with poetry

### PR DESCRIPTION
**Pull request by 42nick**
- Remove unused files.
- Manage test-dependencies via poetry. 

> **_NOTE:_**  The huge diff in the `poetry.lock` is a result of `orion` and `kaleido`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,11 @@ pytest-cov = "^4.1.0"
 numpy = "^1.24.2"
 pytest-benchmark = "^4.0.0"
 
+[tool.poetry.group.test.dependencies]
+matplotlib = "^3.7.0"
+orion = "^0.1.0"  
+kaleido = "0.2.1"
```
Comment (lebrice) [inline pyproject.toml:28]: Why a dev and a test group? Can't we put those in the dev group as well?

**Review by lebrice — APPROVED** [review]
Thanks @42nick !

File: pyproject.toml (hunk 1 above)
Comment (42nick) [reply pyproject.toml]: Yes we can 😄

**Comment by 42nick** [comment]
@lebrice at least I can not merge 😄 Could you please restart the workflow?

**Comment by lebrice** [comment]
Sorry I just fixed the Pre-commit error in #333 , can you merge or rebase? I don't have permission to push to your branch apparently

**Comment by 42nick** [comment]
Nice, I synced with your master :)

**Comment by lebrice** [comment]
Thanks @42nick !

**Pull request merged**